src/common/packager.ts

```diff
@@ -192,7 +194,7 @@ export class Packager {
                                 args = args.concat("--resetCache");
                             }
                             if (runAs === PackagerRunAs.EXPONENT) {
-                                args = args.concat(["--root", ".vscode"]);
+                                args = args.concat(["--root", path.relative(this.projectPath, path.resolve(this.workspacePath, ".vscode"))]);
```

Comment:
I've moved the `exponentIndex.js` file back to `root/.vscode` folder as you suggested, so this path should be correct.
I've tested it with Exponent and it works.
See https://github.com/Microsoft/vscode-react-native/pull/359/commits/f6b41bbfd0a8c1ada2fbcba69f0c37e99ad46369